specification/deviceprovisioningservices/data-plane/Microsoft.Devices/stable/2021-10-01/service.json

```diff
@@ -0,0 +1,1465 @@
+          "description": "Symmetric Key attestation method."
+        }
+      }
+    },
+    "DeviceCapabilities": {
+      "description": "Device capabilities.",
+      "required": [
+        "iotEdge"
+      ],
+      "type": "object",
+      "properties": {
+        "iotEdge": {
+          "description": "If set to true, this device is an IoTEdge device.",
+          "default": false,
+          "type": "boolean"
+        }
+      }
+    },
+    "InitialTwin": {
+      "description": "Initial device twin. Contains a subset of the properties of Twin.",
+      "type": "object",
+      "properties": {
+        "tags": {
+          "$ref": "#/definitions/TwinCollection",
+          "description": "Twin tags."
+        },
+        "properties": {
+          "$ref": "#/definitions/InitialTwinProperties",
+          "description": "Twin desired properties."
+        }
+      }
+    },
+    "ReprovisionPolicy": {
+      "description": "The behavior of the service when a device is re-provisioned to an IoT hub.",
+      "required": [
+        "updateHubAssignment",
+        "migrateDeviceData"
+      ],
+      "type": "object",
+      "properties": {
+        "updateHubAssignment": {
+          "description": "When set to true (default), the Device Provisioning Service will evaluate the device's IoT Hub assignment and update it if necessary for any provisioning requests beyond the first from a given device. If set to false, the device will stay assigned to its current IoT hub.",
+          "default": true,
+          "type": "boolean"
+        },
+        "migrateDeviceData": {
+          "description": "When set to true (default), the Device Provisioning Service will migrate the device's data (twin, device capabilities, and device ID) from one IoT hub to another during an IoT hub assignment update. If set to false, the Device Provisioning Service will reset the device's data to the initial desired configuration stored in the corresponding enrollment list.",
+          "default": true,
+          "type": "boolean"
+        }
+      }
+    },
+    "CustomAllocationDefinition": {
+      "description": "This tells DPS which webhook to call when using custom allocation.",
+      "required": [
+        "webhookUrl",
+        "apiVersion"
+      ],
+      "type": "object",
+      "properties": {
+        "webhookUrl": {
+          "description": "The webhook URL used for allocation requests.",
+          "type": "string"
+        },
+        "apiVersion": {
+          "description": "The API version of the provisioning service types (such as IndividualEnrollment) sent in the custom allocation request. Minimum supported version: \"2018-09-01-preview\".",
+          "type": "string"
+        }
+      }
+    },
+    "TpmAttestation": {
+      "description": "Attestation via TPM.",
+      "required": [
+        "endorsementKey"
+      ],
+      "type": "object",
+      "properties": {
+        "endorsementKey": {
+          "type": "string"
```

Comment:
All properties in the code have a description, but only some generate in the swagger due to the version of Swashbuckle we are using. Will create a PBI to update Swashbuckle, but this is a large task since the next version contains many breaking changes + a new object model. Closing this item for now.

PBI: https://dev.azure.com/msazure/One/_workitems/edit/10911796